sources/core/Stride.Core.CompilerServices/Stride.Core.CompilerServices.csproj

```diff
@@ -18,10 +20,15 @@
 			<Link>Properties\SharedAssemblyInfo.cs</Link>
 		</Compile>
 	</ItemGroup>
-
 	<ItemGroup>
     <PackageReference Include="Microsoft.CodeAnalysis.Analyzers" />
     <PackageReference Include="Microsoft.CodeAnalysis.CSharp" />
+    <PackageReference Include="PolySharp">
+      <PrivateAssets>all</PrivateAssets>
+      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+    </PackageReference>
+
```

Comment:
Where is it used? Is it required at runtime?

Reply:
https://github.com/stride3d/stride/pull/2104/files#diff-015a4d048579b446685ff8c8dbddb5302ff100cd3fbfc1df1cef57c74a4032b6R66

its used here for pattern matching as i got told that i should do it that way

Reply:
Patter-matching is a feature of the language, it doesn't require an additional dependency. Your code seems to compile without Polysharp what does it add?

Reply:
![grafik](https://github.com/stride3d/stride/assets/107197024/2493a141-c468-4c62-833e-0428128d1e03)
System.Index and System.Index.GetOffset

ive changed it to a foreach loop now and added the analyzer removal back in

Reply:
Just to keep a trace of what we discussed on Discord:
1. `netstandard2.0` is a requirement for analyzers/source generators. Nothing we can do about.
2. because of that, certain languages features aren't available (like index/spread operations on collections in C# 12), unless we use a library such as PolySharp to fill the gaps.